dask/bag/text.py

```diff
@@ -61,13 +64,22 @@ def read_text(urlpath, blocksize=None, compression='infer',
     --------
     from_sequence: Build bag from Python sequence
     """
+    files = open_files(urlpath, mode='rt', encoding=encoding,
+                       errors=errors, compression=compression,
+                       **(storage_options or {}))
     if blocksize is None:
-        files = open_files(urlpath, mode='rt', encoding=encoding,
-                           errors=errors, compression=compression,
-                           **(storage_options or {}))
-        blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
+        if files_per_partition is None:
+            blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
+        else:
+            blocks = []
+            for start in range(0, len(files), files_per_partition):
+                block_files = files[start:(start + files_per_partition)]
+                blocks.append(delayed(list)(delayed(files_to_blocks)(block_files)))
```

Comment:
I think that you can replace the entire `files_to_blocks` reference here with `concat`.  My guess is that you don't need to define the function at all.